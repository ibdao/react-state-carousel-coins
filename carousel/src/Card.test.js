import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon"

it("renders without crashing (smoke test)", function (){
    console.log("running card smoke test");
    render(
        <Card 
        caption = { TEST_IMAGES[0].caption }
        src = { TEST_IMAGES[0].src }
        currNum = { 1 } 
        totalNum = { 3 }
        />

    );
});

it("matches snapshot", function (){
    console.log("running card snapshot test");
    const { container } = render(
        <Card 
        caption = { TEST_IMAGES[1].caption }
        src = { TEST_IMAGES[1].src }
        currNum = { 2 } 
        totalNum = { 3 }
        />
    );
    expect(container).toMatchSnapshot();
})