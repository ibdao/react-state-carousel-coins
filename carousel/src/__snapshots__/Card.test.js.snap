// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 2
    </h4>
    <img
      alt="testing image 2"
      class="Card-image"
      src="test2.com"
    />
    <small
      class="Card-small"
    >
      Image 
      2
       of 
      3
      .
    </small>
  </div>
</div>
`;
